### PR TITLE
Support X-Forwarded-Host for tenants

### LINKED ISSUE #1101: Forward Headers for Reverse Proxy
Right now it is not possible to use a reverse proxy or load balancer like NGINX... or it is not documented yet. Do we need Forward Headers Dependency to add?

### PR DESCRIPTION
**Pull request by sebastienros**
Fixes #1101

### CODE REVIEW

File: src/OrchardCore/OrchardCore.Modules/Extensions/RunningShellTableExtensions.cs [hunk 1]
```diff
@@ -1,23 +1,36 @@
-﻿using System;
+using System;
 using Microsoft.AspNetCore.Http;
 using OrchardCore.Environment.Shell;
 
 namespace OrchardCore.Modules
 {
     public static class RunningShellTableExtensions
     {
+        private static string XForwardedHost = "X-Forwarded-Host";
```
Comment (DaRosenberg) [inline src/OrchardCore/OrchardCore.Modules/Extensions/RunningShellTableExtensions.cs]: Should be `const`.

**Comment by PinpointTownes** [comment]
Curious: why is this needed? `ForwardedHeadersMiddleware` (automatically registered by the IIS integration extensions) already takes care of that.

**Comment by sebastienros** [comment]
Do you mean we can just read the Host property and it will take
XForwardedHost if available? Before we were reading "Host" directly so even
if this is the case we had an issue.

**Comment by PinpointTownes** [comment]
> Do you mean we can just read the Host property and it will take XForwardedHost if available?

Yep, that's exactly what I mean.

**Comment by sebastienros** [comment]
Updated

**Pull request merged**